statsmodels/tsa/vector_ar/var_model.py

```diff
@@ -261,85 +280,263 @@ def var_loglike(resid, omega, nobs):
+    params = np.linalg.lstsq(X, Y_sample)[0]
+    resid = Y_sample - np.dot(X, params)
+
+    nobs, neqs = Y_sample.shape
+
+    # MLE estimate of covariance matrix of residuals
+    cov_resid_mle = np.dot(resid.T, resid) / nobs
+    return _get_info_criteria(cov_resid_mle, nobs, neqs, k_ar, k_trend)
+
+def select_order_fit(Y, maxlags=None, ic="aic", trend="c", verbose=False):
+    """
+    Return results from the best lag order according to information criterion
+
+    Parameters
+    ----------
+    Y : array-like
+        The endogenous variables
+    maxlags : int
+        Maximum number of lags to check for order selection, defaults to
+        12 * (nobs/100.)**(1./4), see select_order function
+    ic : {'aic', 'fpe', 'hqic', 'bic', None}
+        Information criterion to use for VAR order selection:
+
+        * aic : Akaike
+        * fpe : Final prediction error
+        * hqic : Hannan-Quinn
+        * bic : Bayesian a.k.a. Schwarz
+    trend : str {"c", "ct", "ctt", "nc"}
+        Available options are:
+
+        * "c" - add constant
+        * "ct" - constant and trend
+        * "ctt" - constant, linear and quadratic trend
+        * "nc" - co constant, no trend
+
+        Note that these are prepended to the columns of X.
+    verbose : bool
+        Print order selection output to the screen. Default is False.
+
+    Returns
+    -------
+    fitted : VARResults
+        A VARResults instance using the best lag order indicated by `ic`.
     """
-    def __init__(self, endog, dates=None, freq=None, missing='none'):
-        super(VAR, self).__init__(endog, None, dates, freq, missing=missing)
+    Y_array = np.asarray(Y)
+    if data_util._is_structured_ndarray(Y_array):
+        Y_array = Y_array.view((float, len(Y_array.dtype)))
+
+    if maxlags is None:
+        maxlags = int(round(12*(len(Y)/100.)**(1/4.)))
+
+    k_trend = util.get_trendorder(trend)
+    ics = defaultdict(list)
+    for k_ar in range(maxlags + 1):
+        # exclude some periods so same nobs used for each lag order
+        res = _estimate_var_ic(Y_array, k_ar, k_trend, trim=maxlags - k_ar,
+                               trend=trend)
+        ics["aic"].append(res[0])
+        ics["bic"].append(res[1])
+        ics["hqic"].append(res[2])
+        ics["fpe"].append(res[3])
+
+    selected_orders = dict((k, mat(v).argmin())
+                           for k, v in ics.iteritems())
+    try:
+        nlags = selected_orders[ic]
+    except:
+        KeyError("ic %s is not understood" % ic)
+    return VAR(Y, nlags).fit(trend=trend)
+
+class VAR(tsbase.TimeSeriesModel):
+    __doc__ = tsbase._tsa_doc % {"model" : _var_model_doc,
+                                 "params" : _var_params_doc,
+                                 "extra_params" : tsbase._missing_param_doc,
+                                 "extra_sections" : _var_reference_doc}
+    def __init__(self, Y, nlags=None, dates=None, names=None, freq=None,
+                 missing='none'):
```

Comment:
refactor: we want design matrix/regressor options in `__init__` not if `fit`
order, trend, exog should be added here and deprecated in `fit`, see ARMA refactoring